### ISSUE TITLE
Future serialize and deserialize

### ISSUE THREAD

**Issue by StoneCypher** [StoneCypher/fsl#1070]
This is an extension set for the previous set of tickets, so that we can close primary properties as complete (because they are.)

Blocked by #999

Future work includes:

* [ ] Identify a storage notation
* [ ] #1007
* [ ] #1008
* [ ] #1010
* [ ] #1056
* [ ] #1057
* [ ] .toString() based on serialize
* [ ] type-augment constructor for consuming serializations
* [ ] type-augment .from/1 for consuming serializations
* [ ] `.from(... , { serialized: foo })`
* [ ] Pack/unpack test cycles for serialization
